### PR TITLE
task: example app updates

### PR DESCRIPTION
**Pull request by luca-gr4vy**
**Description:** Updates the example app to make it more useful when deployed to internal testers. Adds tab navigation with a new Settings screen where you can configure Embed before opening it, persisting this value in the async storage so you don't have to change it again if you reopen the app.

For the token, there are two options:
- Pass the `TOKEN` environment variable with a jwt that was generated manually
- Pass the `GR4VY_PRIVATE_KEY` environment variable, which a base64-formatted version of the private key used to create the token. This will be used by a Node.js script running in a separate thread through the [nodejs-mobile-react-native](https://github.com/Shopify/nodejs-mobile-react-native) library (the Spotify fork, to make it work with simulators on M1 macs and to be able to run the script with arguments). This script will use the `@gr4vy/node` to create the jwt token and pass it back to React Native. This way, any changes to the settings updates the token (for pinned values)

Here's a quick demo: https://drive.google.com/file/d/1PQY1U3YXYYwUXHDHRY1IuY8B3ECEDoFZ/view?usp=sharing

**Ticket:** https://gr4vy.atlassian.net/browse/TA-4604

**Screenshot(s):**

Simulator Screen Shot - iPhone 14 Pro - 2023-05-24 at 15 29 00

Simulator Screen Shot - iPhone 14 Pro - 2023-05-24 at 15 29 08

### CODE REVIEW

File: example/nodejs-assets/build-native-modules-MacOS-helper-script-node.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
...
.../ylvdz3ld5b16ck4gtw6g663w0000gn/T/yarn--1686055109490-0.9487245569392035:/Users/luca/Sites/gr4vy-react-native/example/node_modules/nodejs-mobile-react-native/node_modules/.bin:/Users/luca/.config/yarn/link/node_modules/.bin:/Users/luca/.yarn/bin:/Users/luca/.nvm/versions/node/v16.17.0/libexec/lib/node_modules/npm/bin/node-gyp-bin:/Users/luca/.nvm/versions/node/v16.17.0/lib/node_modules/npm/bin/node-gyp-bin:/Users/luca/.nvm/versions/node/v16.17.0/bin/node_modules/npm/bin/node-gyp-bin:/Users/luca/.rbenv/shims:/Users/luca/.jenv/shims:/Users/luca/.jenv/bin:/Users/luca/Library/Android/sdk/cmdline-tools/latest/bin:/Users/luca/Library/Android/sdk/emulator:/Users/luca/Library/Android/sdk/platform-tools:/Users/luca/.nvm/versions/node/v16.17.0/bin:/Users/luca/.local/bin:/Users/luca/.pyenv/shims:/Users/luca/.pyenv/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin:/Library/Apple/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin
```
Comment (douglaseggleton) [inline example/nodejs-assets/build-native-modules-MacOS-helper-script-node.sh:3]: Should this file be commited with absolute paths `Users/luca` ?
Running `yarn` will change this file to the local users details - so looks like its a generated file.

File: example/nodejs-assets/build-native-modules-MacOS-helper-script-npm.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
.../ylvdz3ld5b16ck4gtw6g663w0000gn/T/yarn--1686055109490-0.9487245569392035:/Users/luca/Sites/gr4vy-react-native/example/node_modules/nodejs-mobile-react-native/node_modules/.bin:/Users/luca/.config/yarn/link/node_modules/.bin:/Users/luca/.yarn/bin:/Users/luca/.nvm/versions/node/v16.17.0/libexec/lib/node_modules/npm/bin/node-gyp-bin:/Users/luca/.nvm/versions/node/v16.17.0/lib/node_modules/npm/bin/node-gyp-bin:/Users/luca/.nvm/versions/node/v16.17.0/bin/node_modules/npm/bin/node-gyp-bin:/Users/luca/.rbenv/shims:/Users/luca/.jenv/shims:/Users/luca/.jenv/bin:/Users/luca/Library/Android/sdk/cmdline-tools/latest/bin:/Users/luca/Library/Android/sdk/emulator:/Users/luca/Library/Android/sdk/platform-tools:/Users/luca/.nvm/versions/node/v16.17.0/bin:/Users/luca/.local/bin:/Users/luca/.pyenv/shims:/Users/luca/.pyenv/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin:/Library/Apple/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin
```
Comment (douglaseggleton) [inline example/nodejs-assets/build-native-modules-MacOS-helper-script-npm.sh:3]: Same again for this file (user paths)

**Comment by douglaseggleton** [comment]
Im getting the following untracked files - we should probably add to `.gitignore`

```
	example/.env.preview
	example/nodejs-assets/nodejs-project/sample-main.js
	example/nodejs-assets/nodejs-project/sample-package.json
```

File: example/.env.example [hunk 3]
```diff
@@ -1,2 +1,3 @@
 GR4VY_ID=spider
-TOKEN=
\ No newline at end of file
+PRIVATE_KEY=
```
Comment (douglaseggleton) [inline example/.env.example:2]: We have the pattern of prefixing env vars with GR4VY_ in other projects

```suggestion
GR4VY_PRIVATE_KEY=
```

**Comment by luca-gr4vy** [comment]
@douglaseggleton my bad, I hadn't merged the changes from main with the react-native upgrade, that's probably why your build failed. Can you try again now? Make sure to run `yarn` again and also `yarn example start --reset-cache` since I also changed the private key env variable name as suggested

**Review by douglaseggleton — APPROVED** [review]
LGTM  👍🏻 - Tested locally and works really well.

**Comment by gr4vy-code** [comment]
:rocket: PR was released in [`v1.1.0`](https://github.com/gr4vy/gr4vy-react-native/releases/tag/v1.1.0) :rocket:

**Pull request merged**